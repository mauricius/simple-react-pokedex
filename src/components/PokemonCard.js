import React from 'react'
import PokemonType from './PokemonType'
import { capitalize } from '../utils'

const PokemonCard = (props) => (
  <div className="max-w-sm overflow-hidden shadow-lg">
    <img className="w-full" src={'https://img.pokemondb.net/artwork/' + props.pokemon.name + '.jpg'} alt="{props.pokemon.name}" />
    <div className="px-6 py-4 border-t-2">
      <div className="font-bold text-xl mb-2">{capitalize(props.pokemon.name)} #{props.pokemon.id}</div>
      <p className="text-grey-darker text-base">
        <b>{capitalize(props.pokemon.name)}</b> is a Pokemon that weights <b>{props.pokemon.weight}</b> pounds. Its base experience is <b>{props.pokemon.base_experience}</b> points.
      </p>
    </div>
    <div className="px-6 py-4">
      {
        props.pokemon.types.map(type => (<PokemonType key={type.slot} type={type} />))
      }
    </div>
  </div>
)

export default PokemonCard