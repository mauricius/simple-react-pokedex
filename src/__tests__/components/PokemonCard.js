import React from 'react'
import PokemonCard from '../../components/PokemonCard'
import renderer from 'react-test-renderer'

describe('test PokemonCard component', function () {

  it('renders correctly', () => {
    const pokemon = {
      id: 1,
      name: 'Bulbasaur',
      weight: 56,
      base_experience: 44,
      types: []
    }

    const component = renderer
      .create(<PokemonCard pokemon={pokemon} />)
      .toJSON()

    expect(component).toMatchSnapshot()
  });
});
