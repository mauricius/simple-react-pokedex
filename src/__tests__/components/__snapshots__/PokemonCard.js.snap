// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PokemonCard component renders correctly 1`] = `
<div
  className="max-w-sm overflow-hidden shadow-lg"
>
  <img
    alt="{props.pokemon.name}"
    className="w-full"
    src="https://img.pokemondb.net/artwork/Bulbasaur.jpg"
  />
  <div
    className="px-6 py-4 border-t-2"
  >
    <div
      className="font-bold text-xl mb-2"
    >
      Bulbasaur
       #
      1
    </div>
    <p
      className="text-grey-darker text-base"
    >
      <b>
        Bulbasaur
      </b>
       is a Pokemon that weights 
      <b>
        56
      </b>
       pounds. Its base experience is 
      <b>
        44
      </b>
       points.
    </p>
  </div>
  <div
    className="px-6 py-4"
  />
</div>
`;
